src/sentry/integrations/source_code_management/issues.py

```diff
@@ -5,40 +5,55 @@
 from typing import Any
 
 from sentry.integrations.mixins.issues import IssueBasicIntegration
+from sentry.integrations.source_code_management.metrics import (
+    SourceCodeIssueIntegrationInteractionEvent,
+    SourceCodeIssueIntegrationInteractionType,
+)
 from sentry.integrations.source_code_management.repository import BaseRepositoryIntegration
 from sentry.models.group import Group
 from sentry.shared_integrations.exceptions import ApiError, IntegrationError
 
 
 class SourceCodeIssueIntegration(IssueBasicIntegration, BaseRepositoryIntegration, ABC):
+    def record_event(self, event: SourceCodeIssueIntegrationInteractionType):
+        return SourceCodeIssueIntegrationInteractionEvent(
+            interaction_type=event,
+            provider_key=self.model.provider,
```

Comment:
`self.integration_name` is only implemented for the RepositoryIntegration class, so we get the provider key via `self.model.provider` instead.